src/libraries/System.Net.NameResolution/src/System/Net/NameResolutionPal.Windows.cs

```diff
@@ -454,6 +454,14 @@ public void RegisterForCancellation(CancellationToken cancellationToken)
                             NetEventSource.Info(@this, $"GetAddrInfoExCancel returned error {cancelResult}");
                         }
                     }
+                    catch (ObjectDisposedException)
+                    {
+                        // Previously, there was a lock on this codepath, but it has been removed since .NET 7.0.
+                        // https://github.com/dotnet/runtime/pull/63904
```

Comment:
```suggestion
```
I do not think it is important to capture what the code was like at some point in the past in comments.